packages/wrangler/src/pages/build.ts

```diff
@@ -483,22 +518,54 @@ We looked for the Functions directory (${basename(
 		);
 	}
 
+	// The build metadata for this build (a hash of the config file it was based on, and the output directory that was used) is usually generated from wrangler.toml
+	// However, frameworks may want to allow users to use a completely different config file, and so may want to fully generate a `_build-metadata.json` file themselves
+	// This is an advanced use case, and should go along with a custom `_worker.bundle` file. The vast majority of users won't need to cutomise this!
+	let buildMetadata: BuildMetadata | undefined;
+
+	if (args.buildOutputDirectory) {
+		const prospectiveBuildMetadataPath = resolvePath(
+			args.buildOutputDirectory,
+			"_build-metadata.json"
+		);
+		if (existsSync(prospectiveBuildMetadataPath)) {
+			const maybeBuildMetadata = parseJSON<{
+				wrangler_config_hash?: string;
+				build_output_directory?: string;
+			}>(await readFile(prospectiveBuildMetadataPath, "utf8"));
+
+			// Ensure we only add known properties to the build metadata
+			buildMetadata =
+				maybeBuildMetadata?.wrangler_config_hash &&
+				maybeBuildMetadata?.build_output_directory
+					? {
+							wrangler_config_hash: maybeBuildMetadata.wrangler_config_hash,
+							build_output_directory: maybeBuildMetadata.build_output_directory,
+					  }
+					: undefined;
```

Comment:
In the case that this file is not valid, should we be logging a warning and/or perhaps falling through to generating the buildMetadata using the logic below?